packages/driver/src/cy/commands/asserting.ts

```diff
@@ -58,28 +96,38 @@ export default function (Commands, Cypress, cy, state) {
     const isCheckingLengthOrExistence = isCheckingExistence || reHaveLength.test(chainers)
 
     const applyChainer = function (memo, value) {
+      logIndex++
       if (value === lastChainer && !isCheckingExistence) {
         // https://github.com/cypress-io/cypress/issues/16006
         // Referring some commands like 'visible'  triggers assert function in chai_jquery.js
         // It creates duplicated messages and confuses users.
         const cmd = memo[value]
 
-        if (_.isFunction(cmd)) {
-          try {
-            return cmd.apply(memo, args)
-          } catch (err: any) {
-            // if we made it all the way to the actual
-            // assertion but its set to retry false then
-            // we need to log out this .should since there
-            // was a problem with the actual assertion syntax
-            if (err.retry === false) {
-              return throwAndLogErr(err)
+        cy.state('onBeforeLog', (log) => {
+          log.set('command', command)
+          log.set('commandLogId', `${assertionIndex}-${logIndex}`)
+        })
+
+        try {
+          if (_.isFunction(cmd)) {
+            try {
+              return cmd.apply(memo, args)
+            } catch (err: any) {
+              // if we made it all the way to the actual
+              // assertion but its set to retry false then
+              // we need to log out this .should since there
+              // was a problem with the actual assertion syntax
+              if (err.retry === false) {
+                return throwAndLogErr(err)
```

Comment:
How can I get in here? I tried a few things, but not entirely clear how to reach this branch.

Reply:
Actually this one does the job: https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/e2e/commands/assertions.cy.js#L379-L384

Reply:
I'm going to be moving this file around and refactoring it a bunch more as part of Detached DOM, so I didn't spend a ton of time commenting every detail. I think the upcoming changes should make this area a bit more straightforward, separating out "command errors" from "execution errors".